openpgp/keys.go

```diff
@@ -111,6 +119,7 @@ func (e *Entity) encryptionKey(now time.Time) (Key, bool) {
 			(!subkey.Sig.FlagsValid && subkey.PublicKey.PubKeyAlgo == packet.PubKeyAlgoElGamal)) &&
 			subkey.PublicKey.PubKeyAlgo.CanEncrypt() &&
 			!subkey.Sig.KeyExpired(now) &&
+			(subkey.Revocation == nil || subkey.Revocation.RevocationReason == nil) &&
```

Comment:
Let's only look at the nil-ness of the revocation. Don't want to skip a revocation that happens to be missing a reason?